### PR TITLE
Fix `unhandledRejection` when aborting a server stream using @bufbuild/connect-node

### LINKED ISSUE #476: `unhandledRejection` when aborting a server stream using `connect-node`
Hello!

I have the impression that with `connect-node`, when you abort a server stream call via an `AbortSignal`, it leads to an `unhandledRejection` systematically.

Is it a bug, or am I using the library incorrectly?

Thank you very much in advance!

**To Reproduce**

```typescript
import {
  createGrpcTransport,
  createPromiseClient,
} from "@bufbuild/connect-node";

import { ElizaService } from "@buf/bufbuild_eliza.bufbuild_connect-es/buf/connect/demo/eliza/v1/eliza_connect.js";

const transport = createGrpcTransport({
  baseUrl: "https://demo.connect.build",
  httpVersion: "2",
})

const client = createPromiseClient(ElizaService, transport);

try {
  const abort = new AbortController();

  for await (const response of client.introduce({name: "Johyn"}, { signal: abort.signal })) {
    console.log('response', response);
    abort.abort();
  }
} catch (e) {
  console.log('error', e);
}
```

```
response IntroduceResponse { sentence: "Hi Johyn. I'm Eliza." }
error ConnectError: [canceled] The operation was aborted
    at connectErrorFromReason (file:///home/johynpapin/Projets/gabie/clienttest/node_modules/@bufbuild/connect-core/dist/esm/connect-error.js:99:20)
    at connectErrorFromNodeReason (file:///home/johynpapin/Projets/gabie/clienttest/node_modules/@bufbuild/connect-node/dist/esm/private/node-error.js:48:12)
    at Promise.reject (file:///home/johynpapin/Projets/ga
bie/clienttest/node_modules/@bufbuild/connect-node/dist/esm/private/node-universal-client.js:285:63)
    at ClientHttp2Stream.h2StreamError (file:///home/johynpapin/Projets/gabie/clienttest/node_modules/@bufbuild/connect-node/dist/esm/private/node-universal-client.js:225:
22)
    at ClientHttp2Stream.emit (node:events:524:35)
    at emitErrorNT (node:internal/streams/destroy:151:8)
    at emitErrorCloseNT (node:internal/streams/destroy:116:3)
    at process.processTicksAndRejections (node:internal/process/task_queues:82:21) {
  rawMessage: 'The operation was aborted',
  code: 1,
  metadata: HeadersList {
    cookies: null,
    [Symbol(headers map)]: Map(0) {},
    [Symbol(headers map sorted)]: null
  },
  details: [],
  cause: undefined
}
node:internal/process/promises:289
            triggerUncaughtException(err, true /* fromPromise */);
            ^

ConnectError: [canceled] The operation was aborted
    at connectErrorFromReason (file:///home/johynpapin/Projets/gabie/clienttest/node_modules/@bufbuild/connect-core/dist/esm/connect-error.js:99:20)
    at connectErrorFromNodeReason (file:///home/johynpapin/Projets/gabie/clienttest/node_modules/@bufbuild/connect-node/dist/esm/private/node-error.js:48:12)
    at Promise.reject (file:///home/johynpapin/Projets/gabie/clienttest/node_modules/@bufbuild/connect-node/dist/esm/private/node-universal-client.js:285:63)
    at ClientHttp2Stream.h2StreamError (file:///home/johynpapin/Projets/gabie/clienttest/node_modules/@bufbuild/connect-node/dist/esm/private/node-universal-client.js:225:22)
    at ClientHttp2Stream.emit (node:events:524:35)
    at emitErrorNT (node:internal/streams/destroy:151:8)
    at emitErrorCloseNT (node:internal/streams/destroy:116:3)
    at process.processTicksAndRejections (node:internal/process/task_queues:82:21) {
  rawMessage: 'The operation was aborted',
  code: 1,
  metadata: Headers {
    [Symbol(headers list)]: HeadersList {
      cookies: null,
      [Symbol(headers map)]: Map(0) {},
      [Symbol(headers map sorted)]: null
    },
    [Symbol(guard)]: 'none'
  },
  details: [],
  cause: undefined
}

Node.js v19.6.0
```

**Environment**

- @bufbuild/connect-node version: 0.7.0
- Node.js version: 19.6.0

**Additional context**

This prevents me from properly testing an API using Jest.

### PR DESCRIPTION
**Pull request by johynpapin**
Hello ! :smile: 

I think I have solved the issue #476. If I understood correctly, the call `sentinel.finally()` returns a new `Promise`, whose failure was never catched.

However, I am not sure that ignoring the error is the right solution. I have the impression that the `sentinel.catch()` earlier in the code makes it safe to do so, but am I wrong?

Thank you very much in advance!

### CODE REVIEW

File: packages/connect-node/src/node-universal-client.ts [hunk 1]
```diff
@@ -321,12 +321,14 @@ function h2Request(
...
-      }
-    });
+    sentinel
+      .finally(() => {
+        session.off("error", sentinel.reject);
+        if (!sessionHolder.keepOpen) {
+          session.close();
+        }
+      })
+      .catch(() => {});
```
Comment (timostamm) [inline packages/connect-node/src/node-universal-client.ts:331]: Could you add a comment?

```suggestion
      .catch(() => {
        // We intentionally swallow sentinel rejection - errors must 
        // propagate through the request or response iterables.
      });
```

**Review by timostamm — COMMENTED** [review]
Great spot!

> However, I am not sure that ignoring the error is the right solution. I have the impression that the `sentinel.catch()` earlier in the code makes it safe to do so, but am I wrong?

Yes, that's correct. Any errors should propagate through the response (body).

File: packages/connect-node/src/node-universal-client.ts (hunk 1 above)
Comment (johynpapin) [reply packages/connect-node/src/node-universal-client.ts]: Of course!

**Comment by johynpapin** [comment]
Sorry for the forced push, I made a mistake in the commit message...

**Review by timostamm — APPROVED** [review]
LGTM!

**Pull request merged**